### PR TITLE
Upgrade astroid to 3.0.0a6

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes
|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |

## Description
Prerequisite for adding Python 3.12 compatibility. 3.12 CI jobs will be added later in #8718.

Skipping news because we have several pre-releases to go before announcing the astroid version to be used for 3.0 final.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -39,7 +39,7 @@ dependencies    = [
     # Also upgrade requirements_test_min.txt.
     # Pinned to dev of second minor update to allow editable installs and fix primer issues,
     # see https://github.com/pylint-dev/astroid/issues/1341
-    "astroid>=3.0.0a5,<=3.1.0-dev0",
+    "astroid @ git+https://github.com/pylint-dev/astroid.git@main",
```
Comment (jacobtylerwalls) [inline pyproject.toml:42]: Waiting for 3.0.0a6 to be released

File: requirements_test_min.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 -e .[testutils,spelling]
 # astroid dependency is also defined in pyproject.toml
-astroid==3.0.0a5  # Pinned to a specific version for tests
+astroid @ git+https://github.com/pylint-dev/astroid.git@main  # Pinned to a specific version for tests
```
Comment (jacobtylerwalls) [inline requirements_test_min.txt:3]: Same

**Comment by jacobtylerwalls** [comment]
Working on patch for the stdlib primer issue (regression in pylint-dev/astroid#2231) 👀

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by jacobtylerwalls** [comment]
Blocked by https://github.com/pylint-dev/astroid/issues/2237.

**Comment by jacobtylerwalls** [comment]
Primer result isn't helpful (same result on #8824). I ran it locally. No change. Something's up on CI. I reopened #8790.

I'll remove the commit that pins to my fix branch. Then I suggest merging this in the next few days. main will fail until we get astroid 3.0.0a7 with the new try node, but I'd like to do that upgrade separately (we already have #7767 for it) and I don't know that we need two separate astroid alphas, one for the hotfix and one for the try node.

@Pierre-Sassoulas your thoughts? :pray:

**Comment by Pierre-Sassoulas** [comment]
Main would fail until we release astroid 3.0.0a7 ?

**Comment by jacobtylerwalls** [comment]
Yeah, if we merge this then main will fail until we upgrade to the next astroid alpha. If that's not okay, then we can do an astroid alpha just for the regression fix. I just didn't want to create extra work.

Or we can hold this PR and merge it into #7767, but I wanted to avoid that because they're two large PRs.

**Comment by Pierre-Sassoulas** [comment]
Let's release a new astroid, this is less work than telling all contributors that the primer also fail on main (it's not that much work, two ``tbump`` and a tag). Should I release asap ?

**Comment by jacobtylerwalls** [comment]
@Pierre-Sassoulas the pin didn't work. The error message from pip is difficult to make sense of. I'll go see if there's an open issue against pip. In any case, should I just rebase #7767 on top of these first three commits and just do a single PR anyway?

**Comment by Pierre-Sassoulas** [comment]
Let's go with a single PR it seems to be the simpler option ? If not do whatever you find more conveniant :D I'm on mobile I can't release astroid 3.0.0a7 right now, I expect to be able to before tommorow

**Comment by jacobtylerwalls** [comment]
Never mind! One file said "astroid" and the other said "astroid.git", and standardizing them fixed the pip resolver.

This should be good to merge.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**